src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -0,0 +1,541 @@
+
+    // set the number of all constraints includes inequality and equality constraints
+    m_QPSolver->data()->setNumberOfConstraints(numberOfAllConstraints);
+
+    m_QPSolver->settings()->setVerbosity(false);
+    m_QPSolver->settings()->setPolish(true);
+
+    m_constraintsMatrix.resize(numberOfAllConstraints, inputSize);
+    m_upperBound.resize(numberOfAllConstraints);
+    m_lowerBound.resize(numberOfAllConstraints);
+    m_gradient.resize(inputSize);
+
+    // set the constant elements of the constraint matrix
+    m_constraintsMatrix(0, 0) = 1;
+    m_constraintsMatrix(1, 1) = 1;
+    m_constraintsMatrix(0, 3) = 1;
+    m_constraintsMatrix(1, 4) = 1;
+    m_constraintsMatrix(6, 2) = 1;
+
+    m_hessianMatrix.resize(m_inputSize, m_inputSize);
+    m_solution.resize(m_inputSize);
+
+    // qpoases
+    m_QPSolver_qpOASES = std::make_unique<qpOASES::SQProblem>(inputSize,
+                                                              m_numberOfConstraints);
+
+    m_QPSolver_qpOASES->setPrintLevel(qpOASES::PL_LOW);
+    m_isFirstTime = true;
+}
+
+bool StepAdaptationController::initialize(const yarp::os::Searchable &config)
+{
+
+    m_inputSize = 5;
+    m_numberOfConstraint = 7;
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "next_zmp_position_weight",  m_zmpPositionWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector";
+        return false;
+    }
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "next_dcm_offset_weight",  m_dcmOffsetWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector";
+        return false;
+    }
+
+    if(!YarpUtilities::getNumberFromSearchable(config, "sigma_weight", m_sigmaWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the number";
+        return false;
+    }
+
+    m_feetExtendedPolygon.resize(2);
+    iDynTree::Polygon foot;
+    iDynTree::Vector4 zmpOffsetLeftFoot;
+    if(!YarpUtilities::getVectorFromSearchable(config, "zmp_offset_leftFoot",  zmpOffsetLeftFoot))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector";
+        return false;
+    }
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "threshold_dcm_error",  m_dcm_ErrorThreshold))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector of DCM error threshold";
+        return false;
+    }
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "threshold_roll_pitch_error",  m_roll_pitch_ErrorThreshold))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector of roll pitch imu error threshold";
+        return false;
+    }
+
+    foot = iDynTree::Polygon::XYRectangleFromOffsets(zmpOffsetLeftFoot(0), zmpOffsetLeftFoot(1),
+                                                     zmpOffsetLeftFoot(2), zmpOffsetLeftFoot(3));
```

Comment:
The offset has a specific meaning. Here it seems you are passing some dimensions instead. Are those the zmp "limits" instead?

Reply:
If I remember correctly they are the offset between the center of foot and zmp!

Reply:
Usually, an offset is a constant quantity added to a varying quantity. Is this the case?

Reply:
Offset also is described as a constant to a common(important) value. for example here the center of foot is an important value for us that zmp has an offset(constant distance) with respect to that.

Reply:
> Offset also is described as a constant to a common(important) value. for example here the center of foot is an important value for us that zmp has an offset(constant distance) with respect to that.

I am not very convinced :stuck_out_tongue_winking_eye: Just try to use a different name

Reply:
Ok! just for your information it's the google translate result for offset :wink:!

> the amount or distance by which something is out of line.
> "these wheels have an offset of four inches"

Reply:
Done!